src/analyzer.ts

```diff
@@ -115,158 +176,94 @@ export class StreamAnalyzer extends Transform {
-    let allHtmlDeps: string[] = [];
-    let allScriptDeps = new Set<string>();
-    let allStyleDeps = new Set<string>();
-
-    let deps: DocumentDeps = this._collectScriptsAndStyles(descriptor);
-    deps.scripts.forEach((s) => allScriptDeps.add(posixPath.resolve(dir, s)));
-    deps.styles.forEach((s) => allStyleDeps.add(posixPath.resolve(dir, s)));
-    if (descriptor.imports) {
-      let queue = descriptor.imports.slice();
-      while (queue.length > 0) {
-        let next = queue.shift();
-        if (!next.href) {
-          continue;
-        }
-        allHtmlDeps.push(next.href);
-        let childDeps = this._getDependenciesFromDescriptor(next, posixPath.dirname(next.href));
-        allHtmlDeps = allHtmlDeps.concat(childDeps.imports);
-        childDeps.scripts.forEach((s) => allScriptDeps.add(s));
-        childDeps.styles.forEach((s) => allStyleDeps.add(s));
-      }
+    let dependencyFilePath = pathFromUrl(this.root, dependencyUrl);
+    if (minimatchAll(dependencyFilePath, this.sourceGlobs)) {
+      logger.debug('dependency is a source file, ignoring...', {dep: dependencyUrl});
+      return;
     }
 
-    return {
-      scripts: Array.from(allScriptDeps),
-      styles: Array.from(allStyleDeps),
-      imports: allHtmlDeps,
-    };
-  }
-
-  _collectScriptsAndStyles(tree: DocumentDescriptor): DocumentDeps {
-    let scripts: string[] = [];
-    let styles: string[] = [];
-    tree.html.script.forEach((script: Node) => {
-      // TODO(justinfagnani): stop patching Nodes in Hydrolysis
-      let __hydrolysisInlined = (<any>script).__hydrolysisInlined;
-      if (__hydrolysisInlined) {
-        scripts.push(__hydrolysisInlined);
-      }
-    });
-    tree.html.style.forEach((style: Node) => {
-      let href = getAttribute(style, 'href');
-      if (href) {
-        styles.push(href);
-      }
-    });
-    return {
-      scripts,
-      styles
-    };
+    let dependencyFile = new File({ path: dependencyFilePath });
+    logger.debug('new dependency found, pushing into dependency stream...', dependencyFile);
+    this._dependenciesStream.push(dependencyFile);
   }
 }
 
-export interface DepsIndex {
-  depsToFragments: Map<string, string[]>;
-  // TODO(garlicnation): Remove this map.
-  // A legacy map from framents to html dependencies.
-  fragmentToDeps: Map<string, string[]>;
-  // A map from frament urls to html, js, and css dependencies.
-  fragmentToFullDeps: Map<string, DocumentDeps>;
-}
 
 class StreamResolver implements Resolver {
+
   analyzer: StreamAnalyzer;
+  deferredFiles = new Map<string, Deferred<string>>();
 
   constructor(analyzer: StreamAnalyzer) {
     this.analyzer = analyzer;
   }
 
+  hasDeferredFile(url: string): boolean {
+    return this.deferredFiles.has(url);
+  }
+
+  hasDeferredFiles(): boolean {
+    return this.deferredFiles.size > 0;
+  }
+
+  deferLoading(url: string, deferred: Deferred<string>): void {
```

Comment:
this function is only called in one place, internal to this class. probably unnecessary